libyul/optimiser/OptimizerUtilities.h

```diff
@@ -45,7 +45,7 @@ void removeEmptyBlocks(Block& _block);
 
 /// Returns true if a given literal can not be used as an identifier.
 /// This includes Yul keywords and builtins of the given dialect.
-bool isRestrictedIdentifier(Dialect const& _dialect, YulName const& _identifier);
+bool isRestrictedIdentifier(Dialect const& _dialect, std::string_view _identifier);
```

Comment:
This change seems unrelated to the ID dispenser. We could merge it right away if you put it in a separate PR.

Reply:
-> PR #15850